### PR TITLE
add conditional parameter for ClusterWS to autoConnect the Socket

### PR DESCRIPTION
**Pull request by ramjak**
### Submitting

- [ ] Bug fix
- [x] Feature
- [ ] Other

### Details
add a conditional parameter for ClusterWS to autoConnect the Socket for issue #53.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -85,6 +88,13 @@ export default class ClusterWS {
     return this.channels[channelName]
   }
 
+  public start(): void {
```
Comment (goriunov) [inline src/index.ts]: I think rename method to  `connect` would be better as in Java and Swift we use `connect`.

File: src/utils/types.ts [hunk 2]
```diff
@@ -17,6 +18,7 @@ export type Options = {
 }
 
 export type Configurations = {
+  autoConnect?: boolean;
```
Comment (goriunov) [inline src/utils/types.ts]: Can you move this line under `url` :)

File: src/index.ts [hunk 3]
```diff
@@ -85,6 +88,13 @@ export default class ClusterWS {
...
+    if (this.hasStarted)
+      console.log('The socket has already been created');
```
Comment (goriunov) [inline src/index.ts]: Can you use `logError` function from utils.

File: src/index.ts [hunk 4]
```diff
@@ -125,5 +135,6 @@ export default class ClusterWS {
       } catch (e) { return logError(e) }
     }
     this.websocket.onerror = (err: ErrorEvent): void => this.events.emit('error', err)
+    this.hasStarted = true
```
Comment (goriunov) [inline src/index.ts]: I think this one can be moved to the top of `create` function

File: src/index.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ export default class ClusterWS {
   private pingTimeout: any
   private pingInterval: number
   private reconnectionAttempted: number = 0
+  private hasStarted: boolean = false
```
Comment (goriunov) [inline src/index.ts]: I think `hasStartedConnection` name suites more in this case.

**Review by goriunov — CHANGES_REQUESTED** [review]
Thank you for pull request there are few small things which needs to be adjusted but overall looks great :)

**Comment by ramjak** [comment]
this one is good?

**Review by goriunov — APPROVED** [review]

**Comment by goriunov** [comment]
Looks good :) I will merge it soon

File: src/index.ts [hunk 6]
```diff
@@ -28,16 +29,18 @@ export default class ClusterWS {
         minInterval: configurations.autoReconnectOptions.minInterval || 1000,
         maxInterval: configurations.autoReconnectOptions.maxInterval || 5000
       } : { attempts: 0, minInterval: 1000, maxInterval: 5000 },
-      encodeDecodeEngine: configurations.encodeDecodeEngine || false
-    }
+      encodeDecodeEngine: configurations.encodeDecodeEngine || false,
+      autoConnect: typeof configurations.autoConnect === 'undefined' ? true : configurations.autoConnect,
```
Comment (goriunov) [inline src/index.ts]: i think you can handle it with just `configurations.autoConnect !== false`

Comment (ramjak) [reply src/index.ts]: hmm, I never thought of that

**Comment by goriunov** [comment]
Thanks looks good :)

**Pull request merged**